homeassistant/components/backup/__init__.py

```diff
@@ -36,22 +43,20 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Backup integration."""
-    hass.data[DOMAIN] = backup_manager = BackupManager(
-        hass, CoreBackupReaderWriter(hass)
-    )
-    await backup_manager.async_setup()
-
     with_hassio = is_hassio(hass)
 
-    async_register_websocket_handlers(hass, with_hassio)
+    if not with_hassio:
+        backup_manager = BackupManager(hass, CoreBackupReaderWriter(hass))
+    else:
+        # pylint: disable-next=import-outside-toplevel, hass-component-root-import
+        from homeassistant.components.hassio.backup import SupervisorBackupReaderWriter
 
-    if with_hassio:
-        if DOMAIN in config:
-            LOGGER.error(
-                "The backup integration is not supported on this installation method, "
-                "please remove it from your configuration"
-            )
-        return True
+        backup_manager = BackupManager(hass, SupervisorBackupReaderWriter(hass))
```

Comment:
We can make this call once outside the if/else block. Just put the BackupReaderWriter instance for core and supervisor cases in a local variable.